app/domain/authentication/authn_oidc/authenticate.rb

```diff
@@ -53,8 +58,7 @@ def new_token(input)
         )
       end
 
-      def conjur_username(request_body)
-        id_token_username_field = "preferred_username"
+      def conjur_username(request_body, id_token_username_field)
         conjur_username = request_body.id_token[id_token_username_field]
         raise IdTokenFieldNotFound, id_token_username_field unless conjur_username&.present?
```

Comment:
As I am not really familiar with the code..  I maybe asking a dummy question
the logic of this POC is only for conjur users that are located in root (and root only) where in the code do you make this assumption?

Reply:
we don't. why should we limit this in the code?

Reply:
it is a limitation we have by design